http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcConfigurationServletListener.java

```diff
@@ -66,7 +67,7 @@ public void contextInitialized(ServletContextEvent sce) {
                 if (is == null) {
                     oidcClientConfiguration = new OidcClientConfiguration();
                 } else {
-                    oidcClientConfiguration = OidcClientConfigurationBuilder.build(is);
+                    oidcClientConfiguration = OidcClientConfigurationBuilder.buildWithoutUnsupportedAttributes(is, servletContext.getInitParameter(JSON_CONFIG_UNSUPPORTED_ATTRIBUTE_PARAM));
```

Comment:
@PrarthonaPaul I've looked more closely into this and have an alternative suggestion for us to consider.

Instead of needing to update Elytron so that it can cope with attributes that are not supported by the default stability level in WildFly, I think we can add handling in `OidcActivationProcessor` in the `elytron-oidc-client` subsystem. In particular, I think we could do the following:

* In `OidcActivationProcessor`, we can get the `WEB-INF/oidc.json` file from the deployment as follows: `deploymentUnit.getAttachment(Attachments.DEPLOYMENT_ROOT).getRoot().getChild("WEB-INF/oidc.json")`.
* That gives us a `VirtualFile` that we could then read the contents of to end up with a `String`.
* We could then add handling to `OidcActivationProcessor#deploy` such that if the stability level isn't the community stability level (i.e., `! deploymentUnit.getStability().enables(Stability.COMMUNITY)`), then if the contents contain the specific attributes that are only available at the community stability level, deployment should fail. Otherwise, the `String` can be used to set the `JSON_CONFIG_CONTEXT_PARAM` servlet context param so that when the `OidcClientConfiguration` is being built, we won't read from the `oidc.json` file a second time, we'll just make use of the String we already have.

WDYT of this approach?

Reply:
I think it is cleaner if in WildFly we can do the stability level validation as stability levels are a WildFly concern not an Elytron concern.

If I am reading this suggestion correctly are we talking about being able to parse from the String directly?

Maybe as a follow up later on another option could be when the Processor in WildFly calls these APIs it can also pass in some predicate for validation?

But for now +1 on constraining to WildFly if we can.

Reply:
If we move the predicate option to a Jira issue we can follow up later and decide if we want to do that.

Reply:
I have updated the code for handling unsupported attributes for deployment config and removed the old code.